### DISCUSSION TITLE
Cmake error @ compile

### DISCUSSION THREAD

**Discussion by ExpGy** [FreeRDP/FreeRDP#10800] [General]
I'm trying to compile the source with usb support, but I keep getting this error message:

cmake -GNinja \
>     -DCMAKE_TOOLCHAIN_FILE=/home/user/freerdp/CMakeLists.txt \
>     -B freerdp-build \
>     -S freerdp \
>     -DCMAKE_BUILD_TYPE=Release \
>     -DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=ON \
>     -DCMAKE_INSTALL_PREFIX=/usr/local/bin/freerdp \
>     -DWITH_SERVER=ON \
>     -DWITH_SAMPLE=ON \
>     -DWITH_PLATFORM_SERVER=OFF \
>     -DUSE_UNWIND=OFF \
>     -DWITH_FFMPEG=OFF \
>     -DWITH_WEBVIEW=OFF \
>     -DCHANNEL_URBDRC=ON \
>     -DWITH_OPENH264=ON \
>     -DWITH_FFMPEG=ON \
>     -DWITH_SWSCALE=ON \
>     -DWITH_DSP_FFMPEG=ON \
>     -DBUILD_SHARED_LIBS=OFF \
>     -DCMAKE_MAKE_PROGRAM=ninja \
>     -DCMAKE_C_COMPILER=/usr/bin/c++

**CMake Error at CMakeLists.txt:26 (project):
  Language 'C' is currently being enabled.  Recursive call not allowed.
Call Stack (most recent call first):
  /usr/local/share/cmake-3.30/Modules/CMakeDetermineSystem.cmake:146 (include)
  CMakeLists.txt:26 (project)**

-- Configuring incomplete, errors occurred!

Any idea for a solution?

**Comment by akallabeth** [comment]
https://github.com/freeRDP/freeRDP/wiki/compilation

`CMakeLists.txt` is not a toolchain file...

**Reply by ExpGy** [reply]
Sorry I'm still a beginner at this!

I'm actually looking for a full version for debian that has usb support, I tried nigthly but it doesn't have it.

**Reply by akallabeth** [reply]
well, just install from the debian repo?

**Reply by akallabeth** [reply]
oh, and nightly has full USB redirection support...

**Comment by ExpGy** [comment]
Omg im simply bind... Thanks

Now xfreerdp connects without an error message with the /usb:auto switch, but it doesn't show the connected usb drive in connected windows desktop.

However, if I use /dvc:urbdrc I get the following error message:

libusb entry returns error.
drdívnc_virtual_channel_event_connected failed with error 4317
Network disconnected!

**Reply by ExpGy** [reply]
I'm really a beginner, sorry. I just checked that the usb drive only works with the /drive: switch.

I fixed the parameters and it works as it should!

**Reply by akallabeth** [reply]
no worries, everyone has to start somewhere
glad it works now.
